docs/search-configuration/custom-dictionaries.md

```diff
@@ -0,0 +1,73 @@
+
+Note that only a file upload is supported for the Japanese user dictionary as it is used by the [kuromoji tokenizer](https://www.elastic.co/guide/en/elasticsearch/plugins/6.3/analysis-kuromoji-tokenizer.html) that splits words up during analysis. Synonyms and stop words are "filters" so multiple files (uploaded and manually entered) can be used for those.
+
+User dictionaries can be configured for [the entire network](admin://network/admin.php?page=search-config) (provided the subsites match the primary site language) or at an individual site level.
+
+## Synonyms
+For better search relevance and results, a dictionary of synonyms can be very useful in the following cases:
+
+- There are a lot of acronyms or other contractions in your content
+- There are common mis-spellings of search terms
+- There are users from different countries who may use different words for the same thing
+- There are sub categories of content that should match a more general search term
+
+For example, you may want search results for the phrase "Checking Accounts" to match content that is titled "Current Accounts" or for common acronyms like "GDP" to match "Gross Domestic Product".
+
+Additionally to cater for variations in language you could create synonyms for American English e.g. "sneakers" to British English e.g. "trainers".
+
+Comma separated lists of words are treated as equivalent. You should have one list of synonyms per line.
+
+```
+sneakers, trainers, footwear, shoes
+foozball, foosball, table football
+CPU, central processing unit
+```
+
+Comma separated words or phrases followed by "=>" will be treated the same as comma separated words or phrases to the right of the "=>" operator but not the other way around.
+
+```
+i-pod, i pod => ipod
+tent => bivouac, teepee
+sea biscuit, sea biscit => seabiscuit
+```
+
+In the above example a search for "tent" will match content containing "bivouac" or "teepee" but a search for "teepee" will only return results containing the word "teepee" specifically.
+
+
+## Stop Words
+Stop words are words that are ignored when analysing content and searching content. Altis uses the standard dictionary of stop words for each supported language. For example in english this includes "it", "of", "and" and so on.
+
+The default list should be adequate in most cases.
+
+A stop words dictionary should contain one word per line:
+
+```
+ignore
+these
+words
+```
+
+
+## Japanese User Dictionary
+Your site's language must be set to Japanese to see this option.
+
+A user dictionary provides a way to control how words are broken up when searching. If there are compound words or phrases specific to this site such as the names of authors or celebrities that users may search for they can be specified here to increase search relevancy.
+
+The syntax for the provided text file should follow the CSV format:
+
+```csv
+text, token 1 ... token n, reading 1 ... reading n, part-of-speech tag
```

Comment:
This example is obscure and not really explained (without knowing Japanese). Is there a way to better describe the format/syntax of the CSV file, specifically what `text`, `token`, `reading` and `part-of-speech tag` refer to?

Reply:
I don't really understand it myself to be honest. Just hoping that between this and the actual Japanese example below it'll make sense to those who need to use it.

Reply:
I wonder if we could pull someone from APAC who has a better handle on it to suggest changes? I hate to leave something like that and then, in the future if we're working with partner devs or people not familiar with the Japanese dictionaries look at our docs and go "huh?" and we're like ¯\_(ツ)_/¯

Reply:
@Kenshino do you understand what the above line of documentation is referring to? Do you think most Japanese speakers would?

Reply:
I don't understand it myself. I don't think Japanese people would @roborourke

Reply:
For reference, here's the full proposed documentation block. I'm not sure it helps that much, but just for context:

## Japanese User Dictionary
Your site's language must be set to Japanese to see this option.

A user dictionary provides a way to control how words are broken up when searching. If there are compound words or phrases specific to this site, such as the names of authors or celebrities that users may search for, they can be specified here to increase search relevancy.

The syntax for the provided text file should follow the CSV format:

```csv
text, token 1 ... token n, reading 1 ... reading n, part-of-speech tag
```

For example:

```csv
東京スカイツリー,東京 スカイツリー,トウキョウ スカイツリー,カスタム名詞
```

Will cause "東京スカイツリー" to be interpreted as "東京" and "スカイツリー" rather than "東京", "スカイ" and "ツリ".

Reply:
If I'm following correctly in the above example:

1. `text` is a string of text that can appear in the website content
2. `token 1...n` is the same text as 1 but with spaces added where the text should be split up
3. `reading 1...n` is the same text as 2 (keeping the spaces) but with kanji written as katakana instead
4. `part-of-speech` indicates whether the text is a noun, adjective, verb and so on

The valid part of speech tags are documented here https://www.kanshudo.com/grammar/parts_of_speech but I can list them in the docs here too. Some examples:

- カスタム名詞 / custom noun
- めいし / 名詞 / noun
- こゆうめし / 固有名詞 / proper noun
- けいしきめいし / 形式名詞 / nominalizer
   > A nominalizer is a word added to a verb or adjective that turns it into a noun phrase, a very common pattern in Japanese, which contains several nominalizers including の, もの, こと etc. See 形式名詞（けいしきめいし).
- どうし /  動詞 / verb
- じょどうし / 助動詞 / auxiliary verb
- ふくごうどうし / 複合動詞 / compound verb
- けいようし / 形容詞 / adjective; い adjective
- けいようどうし / 形容動詞 / な / adjective
  > The origin of the term 形容動詞, which literally means 'verb adjective', is interesting. Originally, な adjectives emerged through the use of なる (become) to a noun (1) - for example 
  > いだい / 偉大
  > なる
  > かわ
  > 川
  > So they were literally 'adjectives formed through addition of a verb'. Over time, 'なる' became just 'な’.
- れんたいし / 連体詞 / other (not い or な) adjective
- ぜんちし / 前置詞 / preposition
- じしょけい / 辞書形 / dictionary form
   > The form that appears in a dictionary. For a verb, this is the plain form of the present tense. For a noun or adjective, it is the uninflected form.
- ふつうけい / 普通形 / plain form(s) of a word
   > The plain forms are the 'casual' or 'short' forms. For example, for the verb 
   > た
   > 食
   > べる (eat), 食べる, 食べた, 食べている etc. are all plain forms.
- ていねいけい / 丁寧形 / polite form(s) of a word
- かつようけい / 活用形 / conjugated form of a word
- しゅうしけい / 終止形 / predicative form
- めいれいけい / 命令形 / imperative form
- しえきけい / 使役形 / causative form
- よだんかつよう / 四段活用 / classical Japanese verb form
- じょうけんけい / 条件形 / conditional form
- じょすうし / 助数詞 / counter
- たいひ / 対比 / comparison
- ひてい / 否定 / negative
- きょうちょう / 強調 / emphasis
- かんりゃくか / 簡略化 / simplification (of a kanji etc.)
- ひっしゃ / 筆者 / cursive (form of a kanji)
- かくにん / 確認 / confirmation
- せつめい / 説明 / explanation
- でんぶん / 伝聞 / hearsay
- もくてき / 目的 / purpose
- ふかひ / 不可避 / inevitability
- けいご / 敬語 / respectful (polite) speech
- ふくじょし / 副助詞 / adverbial particle